task.js

```diff
@@ -0,0 +1,248 @@
+// Клас який описує людину
+class Human {
+    constructor(object){
+      this.name = object.name;
+      this.surname = object.surname;
+      this.age = object.age;
+    }
+    getFullName(){
+      return `${this.name} ${this.surname}`;
+    }
+    setFullName(FullName) {
+      [this.name, this.surname] = FullName.split(' ');
+    }
+  }
+  
+  // -------------------------------------Student------------------------------
+  // Клас який успадковує клас Human
+  class Student extends Human{
+    constructor(info) {
+      super(info);
+      this.marks = info.marks;
+    }
+  
+    // повертає середню оцінку
+    getAverageMark() {
+      return this.marks.reduce((a, b) => (a + b)) / this.marks.length;
+    }
+  
+    // повертає найменшу оцінку
+    getMinMark() {
+      return Math.min(...this.marks);
+    }
+  
+    // повертає максимальну оцінку
+    getMaxMark() {
+      return Math.max(...this.marks);
+    }
+  }
+  
+  // ------------------------------------Teacher-------------------------------
+  
+  class Teacher {
+    constructor(students) {
+      this.students =  students;
+  
+      //Створює об'єкт  де ключи це Імя, а значення це масив з обєктів
+      // this.students = students.reduce((group, arr) => {
+      //   const { name } = arr;
+      //   group[name] = group[name] ?? [];
+      //   group[name].push(arr);
+      //   return group;
+      // }, {}); 
+    }
+    getstudents(){
+      console.table( this.students );
+    }
+  
+    // віддає масив імен студентів відсортованих за оцінками
+    getListOfNamesByAverageMark() {
+      let all_average_mark = [];
+      let name_all_average_mark = [];
+  
+      for (let i_name = 0; i_name < this.students.length; i_name++) {   
+        all_average_mark.push({ name : this.students[i_name].name, average_mark : this.students[i_name].getAverageMark()});
+      }
+      // сортуємо
+      all_average_mark.sort(function (x, y) {
+          return x.average_mark - y.average_mark;
+      });
+      // виводимо тільки І'мя
+      all_average_mark.forEach(element => name_all_average_mark.push(element.name));
+      return name_all_average_mark;
+    }
+  
+    // отримати обєкт студента за його ім'ям
+    getStudentByName(name) {  
+      for (let i in this.students) { 
+        if (name == this.students[i].name) {
+            return this.students[i];
+          }   
+      }
+      return "ERROR";
+    }
+  
+    // заміна або видалення знаючи його ім'я
+    removeAndUpdate(name = false, new_object = false) {
+      let index = false
+      for (let i in this.students) { 
+        if (name === this.students[i].name) {
+            index = i ;
+            break;
+          }   
+      } 
+      if (index) {  
+        if (new_object) {
+          this.students.splice(index, 1,new_object);
+        }
+        else {
+          this.students.splice(index, 1);
+        }
+      }
+      else {
+        console.log("There is no such name ------ ");
+      }
+    }
```

Comment:
Can you explain what exactly the given code mean? In fact you need only the follow lines `this.students.splice(...)`. To find student index use `findIndex`